### ISSUE TITLE
SkipQC parameter not being used

### ISSUE THREAD

**Issue by Glfrey** [nf-core/smrnaseq#208]
### Description of the bug

It seems the pipeline still performs QC when the `--skip_qc` parameter is used

### Command used and terminal output

```console
nextflow run nf-core/smrnaseq --input experiment_info.csv --outdir run_test --genome GRCh37 -profile docker  -r 2.1.0 -c resource.config --skip_qc 

N E X T F L O W  ~  version 22.10.1
Launching `https://github.com/nf-core/smrnaseq` [golden_archimedes] DSL2 - revision: 27514c40c0 [2.1.0]
WARN: Running with Protocol illumina
WARN: Therefore using Adapter: TGGAATTCTCGGGTGCCAAGG
WARN: Clipping 1 bases from R1
WARN: And clipping 2 bases from 3' end


------------------------------------------------------
                                        ,--./,-.
        ___     __   __   __   ___     /,-._.--~'
  |\ | |__  __ /  ` /  \ |__) |__         }  {
  | \| |       \__, \__/ |  \ |___     \`-._,-`-,
                                        `._,._,'
  nf-core/smrnaseq v2.1.0
------------------------------------------------------
Core Nextflow options
  revision           : 2.1.0
  runName            : golden_archimedes
  containerEngine    : docker
  launchDir          : /Users/gillianreynolds/Documents/small_RNAseq/clean_data_run
  workDir            : /Users/gillianreynolds/Documents/small_RNAseq/clean_data_run/work
  projectDir         : /Users/gillianreynolds/.nextflow/assets/nf-core/smrnaseq
  userName           : gillianreynolds
  profile            : docker
  configFiles        : /Users/gillianreynolds/.nextflow/assets/nf-core/smrnaseq/nextflow.config, /Users/gillianreynolds/Documents/small_RNAseq/clean_data_run/resource.config

Input/output options
  input              : experiment_info.csv
  outdir             : run_test

Reference genome options
  genome             : GRCh37
  mirtrace_species   : hsa
  fasta              : s3://ngi-igenomes/igenomes/Homo_sapiens/Ensembl/GRCh37/Sequence/WholeGenomeFasta/genome.fa

Trimming options
  clip_r1            : 1
  three_prime_clip_r1: 2

Skipping pipeline steps
  skip_qc            : true

!! Only displaying parameters that differ from the pipeline defaults !!
------------------------------------------------------
If you use nf-core/smrnaseq for your analysis please cite:

* The pipeline
  https://zenodo.org/badge/latestdoi/140590861

* The nf-core framework
  https://doi.org/10.1038/s41587-020-0439-x

* Software dependencies
  https://github.com/nf-core/smrnaseq/blob/master/CITATIONS.md
------------------------------------------------------
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:INPUT_CHECK:SAMPLESHEET_CHECK                                 -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:CAT_FASTQ                                                     -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTQC_RAW                                       -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:INPUT_CHECK:SAMPLESHEET_CHECK                                 -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:CAT_FASTQ                                                     -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTQC_RAW                                       -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTP                                            -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:INPUT_CHECK:SAMPLESHEET_CHECK                                 -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:CAT_FASTQ                                                     -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTQC_RAW                                       -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTP                                            -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:INPUT_CHECK:SAMPLESHEET_CHECK                                 [  0%] 0 of 1
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:CAT_FASTQ                                                     -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTQC_RAW                                       -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTP                                            -
executor >  local (1)
[6d/f17715] process > NFCORE_SMRNASEQ:SMRNASEQ:INPUT_CHECK:SAMPLESHEET_CHECK (experiment_info.csv)           [  0%] 0 of 1
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:CAT_FASTQ                                                     -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTQC_RAW                                       -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTP                                            -
executor >  local (1)
[6d/f17715] process > NFCORE_SMRNASEQ:SMRNASEQ:INPUT_CHECK:SAMPLESHEET_CHECK (experiment_info.csv)           [  0%] 0 of 1
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:CAT_FASTQ                                                     -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTQC_RAW                                       -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTP                                            -
executor >  local (1)
[6d/f17715] process > NFCORE_SMRNASEQ:SMRNASEQ:INPUT_CHECK:SAMPLESHEET_CHECK (experiment_info.csv)           [  0%] 0 of 1 ✔
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:CAT_FASTQ                                                     -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTQC_RAW                                       -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTP                                            -
executor >  local (1)
[6d/f17715] process > NFCORE_SMRNASEQ:SMRNASEQ:INPUT_CHECK:SAMPLESHEET_CHECK (experiment_info.csv)           [100%] 1 of 1 ✔
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:CAT_FASTQ                                                     -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTQC_RAW                                       [  0%] 0 of 3
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTP                                            [  0%] 0 of 2
executor >  local (2)
[6d/f17715] process > NFCORE_SMRNASEQ:SMRNASEQ:INPUT_CHECK:SAMPLESHEET_CHECK (experiment_info.csv)           [100%] 1 of 1 ✔
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:CAT_FASTQ                                                     -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTQC_RAW                                       [  0%] 0 of 8
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTP                                            [  0%] 0 of 8
executor >  local (2)
[6d/f17715] process > NFCORE_SMRNASEQ:SMRNASEQ:INPUT_CHECK:SAMPLESHEET_CHECK (experiment_info.csv)           [100%] 1 of 1 ✔
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:CAT_FASTQ                                                     -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTQC_RAW                                       [  0%] 0 of 8
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTP                                            [  0%] 0 of 8
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTQC_TRIM                                      -
executor >  local (2)
[6d/f17715] process > NFCORE_SMRNASEQ:SMRNASEQ:INPUT_CHECK:SAMPLESHEET_CHECK (experiment_info.csv)           [100%] 1 of 1 ✔
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:CAT_FASTQ                                                     -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTQC_RAW                                       [  0%] 0 of 8
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTP                                            [  0%] 0 of 8
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTQC_TRIM                                      -
executor >  local (3)
[6d/f17715] process > NFCORE_SMRNASEQ:SMRNASEQ:INPUT_CHECK:SAMPLESHEET_CHECK (experiment_info.csv)           [100%] 1 of 1 ✔
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:CAT_FASTQ                                                     -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTQC_RAW                                       [  0%] 0 of 8
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTP                                            [  0%] 0 of 8
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTQC_TRIM                                      -
executor >  local (4)
[6d/f17715] process > NFCORE_SMRNASEQ:SMRNASEQ:INPUT_CHECK:SAMPLESHEET_CHECK (experiment_info.csv)           [100%] 1 of 1 ✔
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:CAT_FASTQ                                                     -
[b6/4130cd] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTQC_RAW (sampleC)                             [  0%] 0 of 8
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTP                                            [  0%] 0 of 8
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTQC_TRIM                                      -
executor >  local (4)
[6d/f17715] process > NFCORE_SMRNASEQ:SMRNASEQ:INPUT_CHECK:SAMPLESHEET_CHECK (experiment_info.csv)           [100%] 1 of 1 ✔
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:CAT_FASTQ                                                     -
[b6/4130cd] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTQC_RAW (sampleC)                             [  0%] 0 of 8
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTP                                            [  0%] 0 of 8
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:FASTQC_FASTP:FASTQC_TRIM                                      -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:MIRTRACE:MIRTRACE_RUN                                         -
[94/a035c2] process > NFCORE_SMRNASEQ:SMRNASEQ:MIRNA_QUANT:PARSE_MATURE                                      [100%] 1 of 1 ✔
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:MIRNA_QUANT:FORMAT_MATURE                                     [  0%] 0 of 1
[93/c7db6f] process > NFCORE_SMRNASEQ:SMRNASEQ:MIRNA_QUANT:PARSE_HAIRPIN                                     [100%] 1 of 1 ✔
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:MIRNA_QUANT:FORMAT_HAIRPIN                                    [  0%] 0 of 1
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:MIRNA_QUANT:INDEX_MATURE                                      -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:MIRNA_QUANT:INDEX_HAIRPIN                                     -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:MIRNA_QUANT:BOWTIE_MAP_MATURE                                 -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:MIRNA_QUANT:BOWTIE_MAP_HAIRPIN                                -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:MIRNA_QUANT:BAM_STATS_MATURE:SAMTOOLS_SORT                    -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:MIRNA_QUANT:BAM_STATS_MATURE:SAMTOOLS_INDEX                   -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:MIRNA_QUANT:BAM_STATS_MATURE:BAM_STATS_SAMTOOLS:SAMTOOLS_S... -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:MIRNA_QUANT:BAM_STATS_MATURE:BAM_STATS_SAMTOOLS:SAMTOOLS_F... -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:MIRNA_QUANT:BAM_STATS_MATURE:BAM_STATS_SAMTOOLS:SAMTOOLS_I... -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:MIRNA_QUANT:BAM_STATS_HAIRPIN:SAMTOOLS_SORT                   -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:MIRNA_QUANT:BAM_STATS_HAIRPIN:SAMTOOLS_INDEX                  -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:MIRNA_QUANT:BAM_STATS_HAIRPIN:BAM_STATS_SAMTOOLS:SAMTOOLS_... -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:MIRNA_QUANT:BAM_STATS_HAIRPIN:BAM_STATS_SAMTOOLS:SAMTOOLS_... -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:MIRNA_QUANT:BAM_STATS_HAIRPIN:BAM_STATS_SAMTOOLS:SAMTOOLS_... -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:MIRNA_QUANT:EDGER_QC                                          -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:MIRNA_QUANT:SEQCLUSTER_SEQUENCES                              -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:MIRNA_QUANT:BOWTIE_MAP_SEQCLUSTER                             -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:MIRNA_QUANT:MIRTOP_QUANT                                      -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:MIRNA_QUANT:TABLE_MERGE                                       -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:GENOME_QUANT:INDEX_GENOME                                     [  0%] 0 of 1
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:GENOME_QUANT:BOWTIE_MAP_GENOME                                -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:GENOME_QUANT:BAM_SORT_SAMTOOLS:SAMTOOLS_SORT                  -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:GENOME_QUANT:BAM_SORT_SAMTOOLS:SAMTOOLS_INDEX                 -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:GENOME_QUANT:BAM_SORT_SAMTOOLS:BAM_STATS_SAMTOOLS:SAMTOOLS... -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:GENOME_QUANT:BAM_SORT_SAMTOOLS:BAM_STATS_SAMTOOLS:SAMTOOLS... -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:GENOME_QUANT:BAM_SORT_SAMTOOLS:BAM_STATS_SAMTOOLS:SAMTOOLS... -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:MIRDEEP2:MIRDEEP2_PIGZ                                        -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:MIRDEEP2:MIRDEEP2_MAPPER                                      -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:MIRDEEP2:MIRDEEP2_RUN                                         -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:CUSTOM_DUMPSOFTWAREVERSIONS                                   -
[-        ] process > NFCORE_SMRNASEQ:SMRNASEQ:MULTIQC                                                       -
```

### System information

nextflow version 22.10.1
MacBook Pro M1 
Docker container
nfcore version 2.1.0

**Comment by apeltzer** [comment]
I have started working on #210 to fix this too - it has not been used too often as many people always run QC checks on their data (which is probably also a good idea to do ;-)).

**Comment by apeltzer** [comment]
Yes please, maybe also disclose a bit about the data itself as this was not seen / observed so far. Usually takes a couple of hours tops to get through that at least.

**Comment by apeltzer** [comment]
Fixed in PR #202
